pinot-core/src/main/java/org/apache/pinot/core/query/reduce/DistinctDataTableReducer.java

```diff
@@ -86,10 +86,10 @@ public void reduceAndSetResults(String tableName, DataSchema dataSchema,
         //       There's no way currently to get the data types of the distinct columns for empty results
 
         int numColumns = columns.length;
-        DataSchema.ColumnDataType[] columnDataTypes = new DataSchema.ColumnDataType[numColumns];
-        Arrays.fill(columnDataTypes, DataSchema.ColumnDataType.STRING);
-        brokerResponseNative
-            .setResultTable(new ResultTable(new DataSchema(columns, columnDataTypes), Collections.emptyList()));
+        ColumnDataType[] columnDataTypes = new ColumnDataType[numColumns];
+        Arrays.fill(columnDataTypes, ColumnDataType.STRING);
+        brokerResponseNative.setResultTable(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;brokerResponseNative&#39; (index: 3 | type: BrokerResponseNative). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;dataTableMap&#39;. The same parameter type is validated here for example: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java:90. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).